starknet_py/net/base_client.py

```diff
@@ -0,0 +1,187 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Union, Optional, List
+
+from starknet_py.net.client_models import (
+    StarknetBlock,
+    BlockState,
+    Transaction,
+    TransactionReceipt,
+    ContractCode,
+    SentTransaction,
+    InvokeFunction,
+    StarknetTransaction,
+    ContractDefinition,
+    TransactionStatus,
+)
+
+
+@dataclass
+class BlockHashIdentifier:
+    block_hash: int
+    index: int
+
+
+@dataclass
+class BlockNumberIdentifier:
+    block_number: int
+    index: int
```

Comment:
Isn't block number the same as index?

Reply:
So this is in fact something related to tx, right?

Reply:
This is used as a parametr for some client methods since RPC allows querying transactions not only by block hash, but also by pairs like this one.

Reply:
We should create a type
```
Hash = Union[string, int]
TxIdentifier = Union[Hash, BlockHashIdentifier, BlockNumberIdentifier]
```
and reuse it.

Reply:
We have a `Hash` type now, but what about the first question in this thread?

Reply:
Not applicable anymore. Dataclass removed entirely.